contractsV5/KyberNetwork.sol

```diff
@@ -399,16 +404,17 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
             reserve = reserveArr[i];
             //list all reserves that support this token.
             isPayingFees = isFeePayingReserve[address(reserve)];
-            srcAmountWithFee = ((src == ETH_TOKEN_ADDRESS) && isPayingFees) ? srcAmount * (BPS - takerFeeBps) / BPS : srcAmount;
+            //for ETH -> token paying reserve, takerFee is specified in amount
+            srcAmountWithFee = ((src == ETH_TOKEN_ADDRESS) && isPayingFees) ? srcAmount - takerFee : srcAmount;
             rates[i] = reserve.getConversionRate(
                 src,
                 dest,
                 srcAmountWithFee,
                 block.number);
 
-            destAmount = srcAmount * rates[i] / PRECISION;
-             //if token -> ETH swap & reserve is not feeless, deduct fee from destAmount
-            destAmount = (dest == ETH_TOKEN_ADDRESS && isPayingFees) ? destAmount * (BPS - takerFeeBps) / BPS : destAmount;
+            destAmount = srcAmountWithFee * rates[i] / PRECISION;
```

Comment:
this isn't destAmount
maybe call it rateReference?

Reply:
No, rateReference isn't the right term, since it's calculating an amount from the rate.

Reply:
ok
reference amount then
its not dest amount